client/client_update_test.go

```diff
@@ -915,6 +1034,53 @@ func TestUpdateNonRootRemoteCorruptedCanUseLocalCache(t *testing.T) {
 				role:       role,
 			}, testData, false)
 		}
+
+		switch role {
+		case data.CanonicalSnapshotRole:
+			for _, missing := range []string{data.CanonicalRootRole, data.CanonicalTargetsRole} {
+				testData := swizzleExpectations{
+					desc: fmt.Sprintf("snapshot missing root meta checksum"),
+					swizzle: func(s *testutils.MetadataSwizzler, role string) error {
+						return s.MutateSnapshot(func(sn *data.Snapshot) {
+							delete(sn.Meta, missing)
+						})
+					},
+				}
+				testUpdateRemoteCorruptValidChecksum(t, updateOpts{
+					localCache: true,
+					role:       role,
+				}, testData, false)
+			}
+		case data.CanonicalTargetsRole:
+			testData := swizzleExpectations{
+				desc: fmt.Sprintf("target missing delegations data"),
+				swizzle: func(s *testutils.MetadataSwizzler, role string) error {
+					return s.MutateTargets(func(tg *data.Targets) {
+						tg.Delegations.Roles = tg.Delegations.Roles[1:]
+					})
+				},
+			}
+			testUpdateRemoteCorruptValidChecksum(t, updateOpts{
+				localCache: true,
+				role:       role,
+			}, testData, false)
+		case data.CanonicalTimestampRole:
+			testData := swizzleExpectations{
+				desc:       fmt.Sprintf("timestamp missing snapshot meta checksum"),
+				expectErrs: []interface{}{client.ErrMissingMeta{}},
+				swizzle: func(s *testutils.MetadataSwizzler, role string) error {
+					return s.MutateTimestamp(func(ts *data.Timestamp) {
+						delete(ts.Meta, data.CanonicalSnapshotRole)
+					})
+				},
+			}
+			// TODO: this should succeed, because it's an invalid timestamp metadata
+			// and hence should fall back on the the local cached timestamp
```

Comment:
It's because no error occurs in the timestamp validation itself (if one does, we just used the cached timestamp) - the error fails during downloading the snapshot, since there's no checksum.